Week_7.java

```diff
@@ -0,0 +1,166 @@
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+
+/*
+
+Microsoft Interview Coaching Pilot 2018 (MICP 2018)
+Author: Aishna Gupta
+Language: JAVA
+
+************WEEK 6************
+
+Problem description: Design a vending machine.
+Clarify: No more restrictions, just design a general vending machine we are familiar with.
+*/
+
+public class Week_7 {
+	
+	
+	public class Item{
+		private String Item_Id;
+		private String Item_Name;
+		private int Item_Quantity;
+		private int Item_Price;
+		
+		Item(String id, String name, int quantity,int price){
+			Item_Id = id;
+			Item_Name = name;
+			Item_Quantity = quantity;
+			Item_Price = price;
+		}
+		
+		public int getPrice(){
+			return Item_Price;
+		}
+		
+		public void setPrice(int price){
+			 Item_Price = price;
+		}
+		
+		public int getQuantity(){
+			return Item_Quantity;
+		}
+		
+		public void setQuantity(int quantity){
+			 Item_Quantity = quantity;
+		}
+		
+		public String getName(){
+			return Item_Name;
+		}
+		
+		public void setName(String name){
+			 Item_Name = name;
+		}
+
+		public String getId(){
+			return Item_Id;
+		}
+		
+		public void setId(String Id){
+			 Item_Id = Id;
+		}
+		
+		
+	}
+	
+	public class Change{
+		Item item;
+		int change;
+	}
+	
+	public enum CASH{
+		ONE(1), TWO(2), FIVE(5), TEN(10), TWENTY(20), HUNDRED(100), TWO_HUNDRED(200), FIVE_HUNDRED(500);
+		private int value; 
+		
+		private CASH(int value) 
+			{ 
+				this.value = value; 
+			}
+	}
+	
+	public class VendingMachine{
+		int shelves, row, column, totalSales;
+		Item items[][][] = new Item[shelves][row][column];
+		ArrayList<Item> itemsList = new ArrayList<>();
+		int totalCurrency;
```

Comment:
totalCurrency can't really be an int here as you might not have the actual denominations that you need.  If your change is 5 cents, and your vending machine has one quarter in it, totalCurrency would be 25 cents and you would think that you have enough, but you don't have any nickels